### PR TITLE
Rob/tsd80 buildkite ci nix

### PR DESCRIPTION
**Pull request by robcohen**
This PR changes all Nix scripts and adds Buildkite CI (.buildkite/).
Stack2nix is presently not working, so cabal2nix is used (see comment below).

### CODE REVIEW

**Comment by ksaric** [comment]
:tada:

File: shell.nix [hunk 1]
```diff
@@ -1,18 +1,15 @@
-with import ((import ./lib.nix).fetchNixPkgs) { };
-
 let
-  hsPkgs = haskell.packages.ghc822;
+  bootstrap = import <nixpkgs> { };
```
[the remark this answers is not in the document]
Comment (robcohen) [reply shell.nix]: Is there a best-practices resource for creating fetchNixpkgs.nix. It seemed to need updating.

File: CI.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Buildkite
+
+These are the instructions for generating nix expressions for buildkite CI for log-classifier.
+
+```
+$ cd $LOG_CLASSIFIER_ROOT
+$ cabal2nix . > default.nix
+$ nix-shell -p nix-prefetch-git
+nix-shell $ nix-prefetch-git https://github.com/NixOS/nixpkgs.git > nixpkgs.json
```
[the remark this answers is not in the document]
Comment (robcohen) [reply CI.md]: Not clear how to pick a approved rev. So far I think I:
- go to https://github.com/NixOS/nixpkgs
? which branch should be selected?
- click on "Commits"
- click <> of most recent commit?

Or some variation of the above. It's not clear what the best practice is here to me.

[the remark this answers is not in the document]
Comment (cleverca22) [reply CI.md]: the nixos-18.03 branch from https://github.com/nixos/nixpkgs-channels/

[the remark this answers is not in the document]
Comment (robcohen) [reply CI.md]: Done

**Pull request merged**